### PR TITLE
Added validating/defaulter webhooks for OpenTelemetryCollector

### LINKED ISSUE #46: Validating/Defaulter webhooks
As preparation for #45, it would be good to have a simpler, noop Validating/Defaulter webhook. This is useful to test the webhook setup and configuration, and will provide the skeleton for allowing complex CR validation in the future.

### PR DESCRIPTION
**Pull request by jpkrohling**
Closes #46

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

File: api/v1alpha1/opentelemetrycollector_webhook.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+var _ webhook.Defaulter = &OpenTelemetryCollector{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *OpenTelemetryCollector) Default() {
+	opentelemetrycollectorlog.Info("default", "name", r.Name)
+}
+
+// TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
+// +kubebuilder:webhook:verbs=create;update,path=/validate-opentelemetry-io-v1alpha1-opentelemetrycollector,mutating=false,failurePolicy=fail,groups=opentelemetry.io,resources=opentelemetrycollectors,versions=v1alpha1,name=vopentelemetrycollector.kb.io
```
Comment (objectiser) [inline api/v1alpha1/opentelemetrycollector_webhook.go:43]: According to the above comment, should the `delete` be added to the verbs, given the `ValidateDelete` function is included below? And then either way, the TODO comment should probably be removed - or atleast remove the TODO and explain the current decision.

File: config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml [hunk 2]
```diff
@@ -17,6 +17,7 @@ spec:
     - otelcol
     - otelcols
     singular: opentelemetrycollector
+  preserveUnknownFields: false
```
Comment (objectiser) [inline config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml:20]: Is there a reason unknown fields should be allowed?

File: config/crd/kustomization.yaml [hunk 3]
```diff
@@ -8,12 +8,12 @@ resources:
 patchesStrategicMerge:
 # [WEBHOOK] To enable webhook, uncomment all the sections with [WEBHOOK] prefix.
```
Comment (objectiser) [inline config/crd/kustomization.yaml:9]: Should these comment lines be removed once the PR is ready to be merged? (i.e. cleanup)

File: Makefile [hunk 4]
```diff
@@ -81,12 +82,15 @@ generate: controller-gen
...
 # Build the docker image
 docker-build: test
-	docker build . -t ${IMG}
+	docker build -t ${IMG} --build-arg VERSION_PKG=${VERSION_PKG} --build-arg VERSION=${VERSION} --build-arg VERSION_DATE=${VERSION_DATE} --build-arg OTELSVC_VERSION=${OTELSVC_VERSION} .
 
 # Push the docker image
 docker-push:
 	docker push ${IMG}
 
+cert-manager:
```
Comment (objectiser) [inline Makefile:91]: I couldn't find any instructions on when this make target would be called?

File: api/v1alpha1/opentelemetrycollector_webhook.go (hunk 1 above)
Comment (objectiser) [reply api/v1alpha1/opentelemetrycollector_webhook.go:43]: From the manifest it looks like `delete` isn't wanted yet?

Comment (jpkrohling) [reply api/v1alpha1/opentelemetrycollector_webhook.go:43]: The manifest is updated based on the tag above. Your point about the existence of `ValidateDelete` is a good one, I'll add `delete` to the list.

File: config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml (hunk 2 above)
Comment (jpkrohling) [reply config/crd/bases/opentelemetry.io_opentelemetrycollectors.yaml:20]: No idea. To be honest, I haven't given it much thought, as it's required to be `false` for the validation webhook to work. Creating a webhook configuration for a CRD that has this set to `true` will just fail.

File: Makefile (hunk 4 above)
Comment (jpkrohling) [reply Makefile:91]: Will add to the readme.

**Comment by jpkrohling** [comment]
PR updated to address the comments.

File: config/manager/manager.yaml [hunk 5]
```diff
@@ -27,7 +27,7 @@ spec:
         - /manager
         args:
         - --enable-leader-election
-        image: quay.io/opentelemetry/opentelemetry-operator:master
+        image: controller
```
Comment (jpkrohling) [inline config/manager/manager.yaml:30]: This is reverting a previous change. I didn't notice that there is a `kustomization` that replaces the image name.

Comment (objectiser) [inline config/manager/manager.yaml:30]: Why has this changed back from referencing a real image? Is it only used locally?

**Review by objectiser — APPROVED** [review]
LGTM - just one question

File: config/manager/manager.yaml (hunk 5 above)
Comment (objectiser) [reply config/manager/manager.yaml:30]: Ok thanks.

**Pull request merged**